backend/src/schema.graphql

```diff
@@ -318,3 +319,10 @@ type SharedInboxEndpoint {
   id: ID!
   uri: String
 }
+
+type SocialMedia {
+  id: ID!
+  url: String
+  ownedBy: [User]! @relation(name: "OWNED", direction: "IN")
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-04-03T12:27:36Z" title="Wednesday, April 3rd 2019, 2:27:36 pm +02:00">Apr 3, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -310,3 +310,12 @@ type SharedInboxEndpoint {
   id: ID!
   uri: String
 }
+
+type SocialMedia {
+  id: ID!
+  uri: String
```


We have to find a solution for read-only fields. There are attributes that are read-only (disabled flag)  write-only (password) or hidden altogether (privateKey). I think this is a problem of `neo4j-graphql-js`.

If the above problems would not exist I would prefer to have:
```graphql
  addSocialMedia(url: SocialMedia): [SocialMedia]!
```
By the way: @mattwr18 is it not possible to return a SocialMedia object here?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-04-03T16:29:44Z" title="Wednesday, April 3rd 2019, 6:29:44 pm +02:00">Apr 3, 2019</time>_

---


maybe, we were just starting to get this working, ran into some issues and I had to get on with other meetings... I'll just have time to look at it tomorrow I guess

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-04-04T14:02:15Z" title="Thursday, April 4th 2019, 4:02:15 pm +02:00">Apr 4, 2019</time>_

---


We have still the error:
```
Error: Cannot find module 'graphql-constraint-directive'
```

I was not able to find out why … @mattwr18 @roschaefer

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-04-04T14:06:40Z" title="Thursday, April 4th 2019, 4:06:40 pm +02:00">Apr 4, 2019</time>_

---


@roschaefer And I do not understand yet, why you like to have the social media URLs of a person as different nodes? Each URL is unique, because it contains the username of the user on this platform.
Or you like to have the domains of each social media platform in one node and the username stored at the user or as a property in the relation to this social media link?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-04-04T20:47:54Z" title="Thursday, April 4th 2019, 10:47:54 pm +02:00">Apr 4, 2019</time>_

---


Wow! [KISS-principle](https://en.wikipedia.org/wiki/KISS_principle) please! :pray:

The motivation for having `SocialMedia` as extra node is just a matter of design. Before we had social media urls modeled as properties on a user node. But actually, it is a one-to-many relationship and as I said, it's better to use Neo4J's relationships to model relationships, don't use properties. I mean, after all that's what those relationships are for. Plus, it is likely, that our friend `neo4j-graphql-js` understands better what we're doing here if we use our tools in the originally intended way.

The reason why I said "Don't worry about unique constraints" is that we did not think hard enough for use cases why two people might want to share the same URL. Think of a shared organization account as an example.

And don't model social media platforms as enumerable types. Just use URLs for now. We can make it more complex later on.

In general :kiss: kiss: keep-it-simple-stupid